kubetest2-gke/deployer/up.go

```diff
@@ -78,11 +78,21 @@ func (d *deployer) Up() error {
 				args = append(args,
 					"--project="+project,
 					loc,
-					"--machine-type="+d.machineType,
-					"--image-type="+image,
-					"--num-nodes="+strconv.Itoa(d.nodes),
 					"--network="+transformNetworkName(d.projects, d.network),
 				)
+				// A few args are not supported in GKE Autopilot cluster creation, so they should be left unset
+				// when they are not provided via the flags.
+				// https://cloud.google.com/sdk/gcloud/reference/container/clusters/create-auto
+				if d.machineType != "" {
+					args = append(args, "--machine-type="+d.machineType)
+				}
+				if d.nodes != 0 {
+					args = append(args, "--num-nodes="+strconv.Itoa(d.nodes))
+				}
+				if d.imageType != "" {
+					args = append(args, "--image-type="+image)
```

Comment:
This is a bug: 

args = append(args, "--image-type="d.imageType)

Reply:
Fixed.